platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DefaultConfigurationCacheDegradationController.kt

```diff
@@ -57,7 +57,7 @@ internal class DefaultConfigurationCacheDegradationController(
             DegradationDecision(collectTaskDegradationReasons(), collectFeatureDegradationReasons())
         }
 
-    private fun collectTaskDegradationReasons(): Map<Task, List<String>> =
+    private fun collectTaskDegradationReasons(): ImmutableMap<Task, List<String>> =
```

Comment:
❔  If all changes are within implementation details of a Kotlin codebase, aren't `Map`s already immutable?